File: common/src/main/java/com/jmsoftware/maf/common/domain/authcenter/permission/PermissionType.java
package com.jmsoftware.maf.common.domain.authcenter.permission;

import lombok.Getter;
import lombok.extern.slf4j.Slf4j;

/**
 * <h1>PermissionType</h1>
 * <p>Change description here</p>
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com
 * @date 2019-05-25 10:24
 **/
@Slf4j
@Getter
public enum PermissionType {
    /**
     * Controller
     */
    CONTROLLER(Byte.valueOf("0"), "Controller"),
    /**
     * Page
     */
    PAGE(Byte.valueOf("1"), "Page"),
    /**
     * Button
     */
    BUTTON(Byte.valueOf("2"), "Button (API)");

    private final Byte type;
    private final String description;

    PermissionType(Byte type, String description) {
        this.type = type;
        this.description = description;
    }

    public static PermissionType getByType(Byte type) {
        PermissionType permissionType = null;
        PermissionType[] values = PermissionType.values();
        for (PermissionType pt : values) {
            if (pt.getType().equals(type)) {
                permissionType = pt;
            }
        }
        return permissionType;
    }

    /**
     * Get enum by name
     *
     * @param name enum name
     * @return enum
     */
    public static PermissionType getByName(String name) {
        try {
            return PermissionType.valueOf(name);
        } catch (IllegalArgumentException e) {
            log.error("Invalid enum name: {}", name, e);
            return null;
        }
    }
}

File: common/src/main/kotlin/com/jmsoftware/maf/common/domain/authcenter/permission/PermissionType.kt
package com.jmsoftware.maf.common.domain.authcenter.permission

import com.jmsoftware.maf.common.util.logger

/**
 * # PermissionType
 *
 * Change description here
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 6:44 PM
 */
enum class PermissionType(val type: Byte, val description: String) {
    /**
     * Controller
     */
    CONTROLLER(0.toByte(), "Controller"),

    /**
     * Page
     */
    PAGE(1.toByte(), "Page"),

    /**
     * Button
     */
    BUTTON(2.toByte(), "Button (API)");

    companion object {
        private val log = logger()
        fun getByType(type: Byte): PermissionType? {
            var permissionType: PermissionType? = null
            val values = values()
            for (pt in values) {
                if (pt.type == type) {
                    permissionType = pt
                }
            }
            return permissionType
        }

        /**
         * Get enum by name
         *
         * @param name enum name
         * @return enum
         */
        fun getByName(name: String): PermissionType? {
            return try {
                valueOf(name)
            } catch (e: IllegalArgumentException) {
                log.error("Invalid enum name: $name", e)
                null
            }
        }
    }
}
